### PR TITLE
Clean up move_group and moveit_cpp tutorials

### PR DESCRIPTION
**Pull request by MarqRazz**
### Description

- Added `panda_hand_controller` to the loaded controllers in launch files
- ~switched `joint_state_controller` to `joint_state_broadcaster`~
- updated visual_tools text to snake case for better readability
- added collision object example to moveit_cpp tutorial (I am planning on making a PR to remove the demo code from moveit2)

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] While waiting for someone to review your request, please consider reviewing [another open pull request](https://github.com/ros-planning/moveit2/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: doc/move_group_interface/launch/move_group.launch.py
Comment (vatanaksoytezer) [inline doc/move_group_interface/launch/move_group.launch.py:165]: If we want to change `joint_state_controller` to 'joint_state_boradcaster` we also need to change it from `moveit_resources/panda_moveit_config/config/panda_ros_controllers.yaml`. This will affect all the tutorials / demos / applications that is currently using it. I say we wait for it and change in either rolling or galactic.

File: doc/quickstart_in_rviz/launch/demo.launch.py
Comment (vatanaksoytezer) [inline doc/quickstart_in_rviz/launch/demo.launch.py:189]: Same with above comment on `joint_state_controller`

File: doc/moveit_cpp/src/moveit_cpp_tutorial.cpp
Comment (vatanaksoytezer) [inline doc/moveit_cpp/src/moveit_cpp_tutorial.cpp:80]: Any reason we are changing it to not include space but instead `_`?

File: doc/moveit_cpp/src/moveit_cpp_tutorial.cpp
Comment (vatanaksoytezer) [inline doc/moveit_cpp/src/moveit_cpp_tutorial.cpp:119]: Shouldn't this be `Goal pose`?

File: doc/moveit_cpp/src/moveit_cpp_tutorial.cpp
Comment (vatanaksoytezer) [inline doc/moveit_cpp/src/moveit_cpp_tutorial.cpp:273]: That is a nice addition to tutorial, thanks!

File: doc/moveit_cpp/src/moveit_cpp_tutorial.cpp
Comment (vatanaksoytezer) [reply doc/moveit_cpp/src/moveit_cpp_tutorial.cpp:80]: This is question also applicable for other related changes as well.

**Review by vatanaksoytezer — CHANGES_REQUESTED** [review]
I am not sure if we should move to `joint_state_broadcaster` just before the release today. Maybe we can discuss this change on next moveit standup on when and how to do it. 

Rest looks fine and the additional plan is nice! I left a few small nitpicks below. I cannot run this tutorial now correctly, suffering from https://github.com/ros-planning/moveit2/issues/522. We should at least wait for the fix to get merged in moveit2 before merging this.

**Comment by vatanaksoytezer** [comment]
Turns out I cannot run because of `joint_state_broadcaster` is not registered in `moveit_resources`.

File: doc/move_group_interface/launch/move_group.launch.py
Comment (MarqRazz) [reply doc/move_group_interface/launch/move_group.launch.py]: I reverted this change.

File: doc/quickstart_in_rviz/launch/demo.launch.py
Comment (MarqRazz) [reply doc/quickstart_in_rviz/launch/demo.launch.py]: reverted

File: doc/moveit_cpp/src/moveit_cpp_tutorial.cpp
Comment (MarqRazz) [reply doc/moveit_cpp/src/moveit_cpp_tutorial.cpp:119]: This text is what is displayed above the robot to let the user know what section of the tutorial they are in. We publish a `AxisLabeled` on the poses.

File: doc/moveit_cpp/src/moveit_cpp_tutorial.cpp
Comment (MarqRazz) [reply doc/moveit_cpp/src/moveit_cpp_tutorial.cpp:80]: I added the `_` to all of the displayed text because visual_tools puts a ridiculous amount of space between the words
image

**Review by vatanaksoytezer — APPROVED** [review]
I think that enormous space is a RVT bug in ROS2. For now I think this is mergable but I think we should look further into it. I suggest we look into it, I will open an issue on this to track progress on this on RVT repo. I run the tutorials without any problems with the latest version of moveit2. Thanks @MarqRazz !

**Pull request merged**
